src/Microsoft.OData.Core/ODataMetadataFormat.cs

```diff
@@ -161,7 +161,25 @@ public override string ToString()
             ExceptionUtils.CheckArgumentNotNull(messageInfo, "messageInfo");
             ExceptionUtils.CheckArgumentNotNull(messageWriterSettings, "messageWriterSettings");
 
-            throw new ODataException(Strings.General_InternalError(InternalErrorCodes.ODataMetadataFormat_CreateOutputContextAsync));
+            bool isJson = IsJsonMetadata(messageInfo.MediaType);
+
+#if NETSTANDARD2_0
+            if (isJson)
+            {
+                return Task.FromResult<ODataOutputContext>(new ODataMetadataJsonOutputContext(messageInfo, messageWriterSettings));
+            }
+            else
+            {
+                return Task.FromResult<ODataOutputContext>(new ODataMetadataOutputContext(messageInfo, messageWriterSettings));
+            }

```

Comment:
Is the `else` necessary?

Reply:
If clause to create "ODataMetadataJsonOutputContext" => json
else clause to create "ODataMetadataOutputContext"   => xml

So, it's necessary.

---
In reply to: [556528819](https://github.com/OData/odata.net/pull/1966#discussion_r556528819) [](ancestors = 556528819)

Reply:
I think what Clement is saying is that, if it is json, the method will return on line 169 so we could remove the else on line 171 (and associated braces), and just let control fall through to returning xml on line 173.

---
In reply to: [558495394](https://github.com/OData/odata.net/pull/1966#discussion_r558495394) [](ancestors = 558495394,556528819)

Reply:
Oh, I got it.

---
In reply to: [561197162](https://github.com/OData/odata.net/pull/1966#discussion_r561197162) [](ancestors = 561197162,558495394,556528819)